parse_date.re

```diff
@@ -707,25 +707,50 @@ static const timelib_tz_lookup_table* abbr_search(const char *word, timelib_long
 	return NULL;
 }
 
+#define TIMELIB_TP_VALUE(tp) tp->gmtoffset - (tp->type * 3600)
+
 static timelib_long timelib_lookup_abbr(char **ptr, int *dst, char **tz_abbr, int *found)
 {
 	char *word;
 	char *begin = *ptr, *end;
 	timelib_long  value = 0;
 	const timelib_tz_lookup_table *tp;
 
+	int has_dash = 0;
 	while (**ptr != '\0' && **ptr != ')' && **ptr != ' ') {
-		++*ptr;
+		if(*((*ptr)++) == '-'){
+			has_dash = 1;
+		}
 	}
 	end = *ptr;
 	word = timelib_calloc(1, end - begin + 1);
 	memcpy(word, begin, end - begin);
 
 	if ((tp = abbr_search(word, -1, 0))) {
-		value = tp->gmtoffset;
+		value = TIMELIB_TP_VALUE(tp);
 		*dst = tp->type;
-		value -= tp->type * 3600;
 		*found = 1;
+	} else if(has_dash) {
+		char *cpy = timelib_calloc(1, end - begin + 1);
+		char *pos = NULL;
+
+		memcpy(cpy, word, end - begin);
+
+		while((pos = strrchr(cpy, '-')) != NULL){
+			*pos = '\0';
+			if((tp = abbr_search(cpy, -1, 0))){
+				value = TIMELIB_TP_VALUE(tp);
```

Comment:
This is now turned into a much more expensive operation, for a minor edge case. I don't think I want to make this change.

Reply:
I tried to minimize the cost by checking for the dash while looking ahead, to only do that if we're in the worst case scenario.
The best-case scenario is if the timezone is correct (which involves no extra cost other than the if condition for the dash)

The worst case scenario is when the user inputs a date that has a lot of trailing data after the timezone, which requires unrolling all the dashes
I can try to miminize it further by counting the number of dashes in the while loop and abort early if it's more than the maximum allowed.